### PR TITLE
Provide credentials to nightly WPT sync job

### PR DESCRIPTION
**Pull request by jdm**
This will give us more useful information by actually creating a PR with the results of syncing with upstream. We can choose to disregard it without any issue.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/783)

### CODE REVIEW

File: buildbot/master/files/config/environments.py [hunk 1]
```diff
@@ -152,4 +153,5 @@ def without(self, to_unset):
     'AWS_ACCESS_KEY_ID': S3_UPLOAD_ACCESS_KEY_ID,
     'AWS_SECRET_ACCESS_KEY': S3_UPLOAD_SECRET_ACCESS_KEY,
     'GITHUB_HOMEBREW_TOKEN': GITHUB_HOMEBREW_TOKEN,
+    'WPT_SYNC_TOKEN': WPT_SYNC_PR_CREATION_TOKEN,
```
Comment (aneeshusa) [inline buildbot/master/files/config/environments.py]: Why are we putting this in the `upload_nightly` environment? It sounds like this is being run as a separate step, so we should use a separate environment (e.g. I find it unlikely this needs AWS access), and use the environment combining feature with `+` if necessary.

File: buildbot/master/init.sls [hunk 2]
```diff
@@ -38,6 +38,7 @@ deploy-{{ common.servo_home }}/buildbot/master:
     - context:
         common: {{ common }}
         buildbot_credentials: {{ pillar['buildbot']['credentials'] }}
+        wpt_sync_credentials: {{ pillar['wpt-sync'] }}
```
Comment (aneeshusa) [inline buildbot/master/init.sls]: I'd prefer if these credentials got moved under `buildbot:credentials` so this line is not needed. The test credential would be removed from the `.travis/test_pillars/wpt-sync.sls` file and moved into the `.travis/test_pillars/buildbot/master.sls`.

Comment (jdm) [reply buildbot/master/init.sls]: It's also used by https://github.com/servo/saltfs/blob/master/upstream-wpt-webhook/files/config.json#L2 so it won't actually be removed.

Comment (aneeshusa) [reply buildbot/master/init.sls]: I didn't realize that, keeping the current setup makes sense to me then.

File: buildbot/master/files/config/environments.py (hunk 1 above)
Comment (jdm) [reply buildbot/master/files/config/environments.py]: Fixed.

**Comment by aneeshusa** [comment]
Thanks!
@bors-servo r+

**Comment by aneeshusa** [comment]
Looks like there is a lint error:
```
[ FAIL ] Trailing whitespace found on files and lines:
         ./buildbot/master/files/config/environments.py:156:    'WPT_SYNC_TOKEN': WPT_SYNC_PR_CREATION_TOKEN,----
```

**Pull request merged**